pkgs/applications/networking/instant-messengers/zoom-us/default.nix

```diff
@@ -66,15 +74,27 @@ in mkDerivation {
       runHook postInstall
     '';
 
-  postInstall = (makeDesktopItem {
-    name = "zoom-us";
-    exec = "$out/bin/zoom-us %U";
-    icon = "$out/share/zoom-us/application-x-zoom.png";
-    desktopName = "Zoom";
-    genericName = "Video Conference";
-    categories = "Network;Application;";
-    mimeType = "x-scheme-handler/zoommtg;";
-  }).buildCommand + ''
+  postInstall = ''
+    mkdir -p $out/share/{applications,appdata,icons}
+
+    # Desktop File
+    cp ${desktopIntegration}/us.zoom.Zoom.desktop $out/share/applications
+    substituteInPlace $out/share/applications/us.zoom.Zoom.desktop \
+        --replace "Exec=zoom" "Exec=$out/bin/zoom-us"
+
+    # Appdata
+    cp ${desktopIntegration}/us.zoom.Zoom.appdata.xml $out/share/appdata
+
+    # Icons
+    icon_sizes=("64" "96" "128" "256")
+
+    for icon in ''${icon_sizes[*]}
```

Comment:
How about 

```bash
for size in 64 96 128 256; do
  ...
done
```

Reply:
I think you'll find https://github.com/NixOS/nixpkgs/pull/65704/commits/b46a365a710fd9b1cb8ca95992176e87728d1fbc more readable.